db/sequence_allocator.go

```diff
@@ -229,6 +225,7 @@ func (s *sequenceAllocator) nextSequenceGreaterThan(ctx context.Context, existin
 	if targetSequence <= s.max {
 		releaseFrom := s.last + 1
 		s.last = targetSequence
+		s.dbStats.LastSequenceAssignedValue.Set(int64(targetSequence))
```

Comment:
Does it make sense to abstract this into a function so if `s.last` is set then this is always incremented in case we move this?

Should `s.last` be named `s._last` to indicate you need a lock for read/write?

Reply:
I started to do this refactor and it unraveled a slight difference between `s.last`'s value and an actually allocated/assigned sequence (reported via the stat), which aren't always the same - in cases of releasing unused sequences or allocating a new batch without allocating a sequence.

It made stuff more complicated for the special cases where we don't want to update the stat, and it actually broke stuff in my attempts.

I'm 50/50 about naming as well. The mutex guards most of the whole struct really, and renaming `_last`, `_max`, `_sequenceBatchSize`, and `_lastSequenceReserveTime` becomes pretty noisy in diff and readability.